src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -327,7 +344,8 @@ public Variables GetVariableValues(Schema schema, Variables variables, Inputs in
 
         public object GetVariableValue(Schema schema, Variable variable, object input)
         {
-            var type = schema.FindType(variable.Type.Name);
+            var typeNameFormat = variable.Type.IsList ? "[{0}]" : "{0}";
+            var type = schema.FindType(string.Format(typeNameFormat, variable.Type.Name));
```

Comment:
I had to do something like this too.  I think this "full name" should probably be moved to the `Variable` type itself.